### PR TITLE
!!! BUGFIX: Revert adjustment of ContentCollection Constraint adjustment

### PR DESCRIPTION
**Pull request by mficzel**
In https://github.com/neos/neos-development-collection/pull/2715 the Constraints of the ContentCollection were changed to allow Content instead of denying Documents.

This caused trouble because the order the constraints of abstract nodes are applied is not defined
and thus derived collections that forbade Neos.Neos:Content but allowed a custom NodeType
ran into trouble.

Reverts the following commits:
- 0b88d4b8d9ceb5ed1e97dcd1056c8c3443cc98db of pr #2715
- 24a785d650308fac43a46cca90916aac31e8eb0e of pr #3163
- 61123e509e0d5994772c978e8c0e4566e7d4bd2c of pr #3178

Resolves: #3212

### CODE REVIEW

File: Neos.Neos/Configuration/NodeTypes.ContentCollection.yaml [hunk 1]
```diff
@@ -9,5 +9,6 @@
     inlineEditable: true
   constraints:
     nodeTypes:
-      '*': false
-      'Neos.Neos:Content': true
+      'Neos.Neos:Document': false
```
Comment (bwaidelich) [inline Neos.Neos/Configuration/NodeTypes.ContentCollection.yaml:12]:
```suggestion
      # Note: Reverting the constraints to explicitly allow content nodes instead won't work, see https://github.com/neos/neos-development-collection/issues/3212
      'Neos.Neos:Document': false
```

**Comment by mficzel** [comment]
I did not revert the adjustment of the test nodetypes. Those should be fine but i suggest to wait for the tests.

**Review by bwaidelich — COMMENTED** [review]
Great, thank you.
Do you think we should add that comment?

**Review by bwaidelich — APPROVED** [review]
<3

**Pull request merged**